File: app/src/main/java/com/example/cats/db/ShowDao.java
package com.example.cats.db;

import android.arch.persistence.room.*;
import io.reactivex.Single;

import java.util.List;

@Dao
public interface ShowDao {
    @Query("SELECT * FROM favorite_cats")
    Single<List<FavoriteCats>> getAllFavoriteCats();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(FavoriteCats favoriteCats);

    @Delete
    void remove(FavoriteCats favoriteCats);

    @Query("SELECT count(*) FROM favorite_cats where id LIKE :id")
    int isFavouriteCats(long id);
}

File: app/src/main/java/com/example/cats/db/ShowDao.kt
package com.example.cats.db

import android.arch.persistence.room.*
import io.reactivex.Flowable
import io.reactivex.Single

@Dao
interface ShowDao {
    @Query("SELECT * FROM favorite_cats")
    fun allFavoriteCats(): Flowable<List<FavoriteCats>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(favoriteCats: FavoriteCats)

    @Delete
    fun remove(favoriteCats: FavoriteCats)

    @Query("SELECT count(*) FROM favorite_cats where id LIKE :id")
    fun isFavouriteCats(id: Long): Int
}

